### PR TITLE
fix: reprs had missing information

### LINKED ISSUE #200: [BUG] Repr issues
The integer repr is missing name/label, and label gets repeated even if it's just picking it up from name for the other axes, like Regular.

### PR DESCRIPTION
**Pull request by amangoel185**
Fixes #200

### CODE REVIEW

**Comment by henryiii** [comment]
Could you add a test for the repr to make this doesn't happen again?

**Comment by amangoel185** [comment]
Sure

**Comment by amangoel185** [comment]
Sure, I'll do that.

**Comment by amangoel185** [comment]
I've added the tests, @henryiii. You can merge it if it looks good to you.

File: src/hist/axis/__init__.py [hunk 1]
```diff
@@ -212,8 +212,8 @@ def __init__(
         *,
         name: str = "",
         label: str = "",
-        growth: bool = False,
         metadata: Any = None,
+        growth: bool = False,
```
Comment (henryiii) [inline src/hist/axis/__init__.py:216]: Not sure why these got reordered, but due to the "*" above, order does not affect users, so it's safe. :)

**Review by henryiii — APPROVED** [review]

File: src/hist/axis/__init__.py (hunk 1 above)
Comment (amangoel185) [reply src/hist/axis/__init__.py:216]: Oh, I reordered them since I saw that order in `Regular`.

**Pull request merged**